src/S3/StreamWrapper.php

```diff
@@ -325,7 +329,7 @@ public function rmdir($path, $options)
      */
     public function dir_opendir($path, $options)
     {
-        $this->clearStatInfo();
+        $this->openedPath = $path;
```

Comment:
By tracking the opened path, we can reopen the path when dir_rewinddir is called (used in things like the Symfony Finder).